Add on_kill  equivalent to Databricks SQL Hook to cancel timed out queries

R7L208 opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

related: #42115 

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

The Databricks Provider did not implement a mechanism to cancel SQL queries submitted by DatabricksSqlHook. This led to data quality issues, where Airflow would report a cancellation due to timeout; however, the corresponding SQL query would continue to run on Databricks.

This PR uses threading to cancel SQL queries submitted by DatabricksSqlHook.run() once the timeout is exceeded.

<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

Commit:
add on_kill override to databricks workflow operator

Commit:
add on_kill override to databricks workflow operator

Commit:
on_kill equivalent for DatabricksSqlOperator

Commit:
on_kill equivalent for DatabricksSqlOperator

Commit:
add tests for create_timeout_thread

Commit:
add tests for create_timeout_thread

Commit:
add note for on_kill in DatabricksCopyIntoOperator

Commit:
add note for on_kill in DatabricksCopyIntoOperator

Commit:
chore: static checks

Commit:
chore: static checks

Commit:
remove changes for databricks_workflow.py for PR isolated to databricks_sql

Commit:
remove changes for databricks_workflow.py for PR isolated to databricks_sql

Commit:
remove file accidently picked up rebasing with main

Commit:
Revert "remove file accidently picked up rebasing with main"

This reverts commit 13287fcc63cfe2f3df548c287bd6386007520167.

Commit:
remove file from tracking

Commit:
adding updated test_databricks_workflow file

Commit:
add indent to execution_timeout param in docstring of DatabricksSqlHook.run

Commit:
add indent to execution_timeout param in docstring of DatabricksSqlHook.run

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

Lee-W commented on airflow/providers/databricks/hooks/databricks_sql.py:
```diff
@@ -248,7 +265,22 @@ def run(
-                    self._run_command(cur, sql_statement, parameters)  # type: ignore[attr-defined]
+                    t = create_timeout_thread(cur, execution_timeout)
+
+                    # todo; adjust this to make testing easier
```
```suggestion
                    # TODO: adjust this to make testing easier
```

Lee-W commented on airflow/providers/databricks/hooks/databricks_sql.py:
```diff
@@ -248,7 +265,22 @@ def run(
+                    # todo; adjust this to make testing easier
+                    try:
+                        self._run_command(cur, sql_statement, parameters)  # type: ignore[attr-defined]
+                    except Exception as e:
```
Is it possible for us to carry a narrower exception?

R7L208 replied:
I don't believe it is possible since we want every error message to be propagated back; however, if it's due to the query being canceled by the thread `t`, we want to create a specific timeout error message instead of only propagating the error message from Databricks.

Lee-W replied:
sounds good 👍

Lee-W commented on airflow/providers/databricks/hooks/databricks_sql.py:
```diff
@@ -248,7 +265,22 @@ def run(
+                        self._run_command(cur, sql_statement, parameters)  # type: ignore[attr-defined]
+                    except Exception as e:
+                        if t is None or t.is_alive():
+                            raise AirflowException(f"Error running statement: {sql_statement}. {str(e)}")
```
Let's create a new exception (inherited AirflowException) for this one instead of using `AirflowExecption`

R7L208 replied:
Will create AirflowTaskExecutionError

Lee-W commented on airflow/providers/databricks/hooks/databricks_sql.py:
```diff
@@ -248,7 +265,22 @@ def run(
+                        if t is None or t.is_alive():
+                            raise AirflowException(f"Error running statement: {sql_statement}. {str(e)}")
+                        else:
+                            raise AirflowException(
```
same here

R7L208 replied:
I will create an AirflowTaskExecutionTimeout for this one

R7L208 committed:
Update airflow/providers/databricks/hooks/databricks_sql.py

Co-authored-by: Wei Lee <weilee.rx@gmail.com>

R7L208 committed:
Update airflow/providers/databricks/hooks/databricks_sql.py

Co-authored-by: Wei Lee <weilee.rx@gmail.com>

Commit:
add new exceptions

Commit:
add new exceptions

Commit:
add new exceptions

Commit:
add new exceptions

Commit:
fix typing warning

Commit:
fix typing warning

Commit:
fix ruff error

Commit:
fix ruff error

Commit:
fix ruff error

Commit:
fix ruff error

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

uranusjr commented on airflow/providers/databricks/hooks/databricks_sql.py:
```diff
@@ -35,7 +37,12 @@
+    AirflowException,
+    AirflowProviderDeprecationWarning,
+    AirflowTaskExecutionError,
+    AirflowTaskExecutionTimeout,
```
You can’t do this because providers have backward compatibility obligations and need to work on older Airflow versions. Since these are only used in this provider anyway, the exceptions can simply be declared in the provider instead.

Commit:
move new exceptions to Databricks provider

Commit:
move new exceptions to Databricks provider

Commit:
ruff formatting

Commit:
ruff formatting

Commit:
add tests for new exceptions

Commit:
add tests for new exceptions

Commit:
chore - static checks

Commit:
chore - static checks

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

kaxil committed:
Simplify code for recent dbt provider change (#42840)

Just simplified redundant code from https://github.com/apache/airflow/pull/42737

harjeevanmaan committed:
- Fixes #42432 (#42770)

- Added a unit test

Lee-W commented on airflow/providers/databricks/exceptions.py:
```diff
@@ -0,0 +1,32 @@
+
+
+class AirflowTaskExecutionTimeout(AirflowTaskExecutionError):
+    """Raised when a task execution times out."""
```
Instead of naming it as Airflow, should we make it more specific to Databricks?

R7L208 replied:
I'm not opposed, but I've been trying to follow similar patterns and styles to other providers. I've not seen provider-specific exceptions, which primarily just use AirflowException directly.

@Lee-W Let me know your thoughts on this and happy to adjust the code according

Lee-W replied:
https://github.com/apache/airflow/blob/61aadddf5158ee30b7f375dbd947b16e4afa3942/providers/src/airflow/providers/amazon/aws/exceptions.py#L26

I think many providers actually use their own name. You can take a look at the aws one and maybe others as well. Thanks :)

Lee-W commented on airflow/providers/databricks/hooks/databricks_sql.py:
```diff
@@ -248,7 +269,24 @@ def run(
+                        else:
+                            raise AirflowTaskExecutionTimeout(
+                                f"Timeout threshold exceeded for query: {sql_statement} was cancelled."
+                            )
```
```suggestion
                        if t is None or t.is_alive():
                            raise AirflowTaskExecutionError(
                                f"Error running statement: {sql_statement}. {str(e)}"
                            )
                        raise AirflowTaskExecutionTimeout(
                            f"Timeout threshold exceeded for query: {sql_statement} was cancelled."
                        )
```

bugraoz93 committed:
AIP-84 Migrate views /object/historical_metrics_data to Fast API (#42629)

* Include object router and migrate objects/historical_metrics_data to FastAPI

* Use pyfixture session

* Include provide_session for SQLite session thread issue

* Make method async and remove unused session from tests

* Include return type to generate proper openapi spec

* Object definition to Dashboard definition, make safe date Annotated parameter in parameters.py

* Include pydantic model for Dashboard Historical Metric Data and convert response to HistoricalMetricDataResponse

* Fix conflict and rerun pre-commit hooks

* Rename data variable to historical metrics for consistency

* Include object router and migrate objects/historical_metrics_data to FastAPI

* Object definition to Dashboard definition, make safe date Annotated parameter in parameters.py

* Drop data from method name to prevent double Data in the name of objects

* Variable name change to prevent shadow naming with method name

* Resolve conflicts and rebase again

* Resolve conflicts and rebase

tirkarthi committed:
Add support to filter by last dagrun state in UI. (#42779)

* Add support to filter by last dagrun state in UI.

* Fix lint errors.

* Fix lint errors.

* Fix PR comments over null checks and query parameter to be last_dag_run_state.

Lee-W committed:
feat(datasets): make strict_dataset_uri_validation default to True (#41814)

* feat(datasets): make strict_dataset_uri_validation default to True

* docs(assets): add uri scheme restrict instruction for AIP-60

* feat(assets): add an error logging to URI that is not AIP-60 compliant

* Better test string

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

gopidesupavan committed:
Standard provider bash operator (#42252)

romsharon98 committed:
Fix mark as success when pod fails while fetching log (#42815)

* fix: always defer once more after log fetching to ensure pod completion is handled

* add tests

* removing print

---------

Co-authored-by: Jean-Eudes Peloye <peloye.je@gmail.com>

jbampton committed:
Fix spelling; `Airlfow` -> `Airflow` (#42855)

rawwar committed:
Docs: Add Template field related info for python operators (#42847)

closes #39584

This PR adds templating information related to Python operators in How-to Guide

pierrejeambrun committed:
AIP-84 Get Variable (#42834)

Handle redacted value for Variables

dstandish committed:
Move test of DagRun.update_state to better place (#42845)

Previously this lived in test_scheduler_job.py

It only really tested the behavior of DagRun.update_state.

As far as I can tell, it checks that if you null out the state on a TI of a finished dag, and then you call ``update_state``, then the DR will be set to running.

mobuchowski committed:
Revert "Remove `sqlalchemy-redshift` dependency from Amazon provider (#42830)" (#42864)

This reverts commit 3b4c73a644f40013927fb72227dfe018dc47a97f.

eladkal committed:
Prepare docs for Oct 1st adhoc wave of providers (#42862)

* Prepare docs for Oct 1st adhoc wave of providers

* Update

* update amazon

dannyl1u committed:
fix more PT004 PyDocStyle checks (#42841)

kaxil committed:
Revert Asset to Dataset for Core Extension Doc (#42867)

This PR reverts Core Extension doc change (https://airflow.apache.org/docs/apache-airflow-providers/core-extensions/index.html) from https://github.com/apache/airflow/pull/41348

Since those docs only work on Airflow stable, we can only change this after 3.0

ashb committed:
Split providers out of the main "airflow/" tree into a UV workspace project (#42505)

This is only a partial split so far. It moves all the code and tests, but
leaves the creation of `core/` to a separate PR as this is already large
enough.

In addition to the straight file rename the other changes I had to make here
are:

- Some mypy/typing fixes.

  Mypy can be fragile about what it picks up when, so maybe some of those
  changes were caused by that. But the typing changes aren't large.

- Improve typing in common.sql type stub

  Again, likely a mypy file oddity, but the types should be safe

- Removed the `check-providers-init-file-missing` check

  This isn't needed now that airflow/providers shouldn't exist at all in the
  main tree.

- Create a "dev.tests_common" package that contains helper files and common
  pytest fixtures

  Since the provider tests are no longer under tests/ they don't automatically
  share the fixtures from the parent `tests/conftest.py` so they needed
  extracted.

  Ditto for `tests.test_utils` -- they can't be easily imported in provider
  tests anymore, so they are moved to a more explicit shared location.

In future we should switch how the CI image is built to make better use of UV
caching than our own approach as that would remvoe a lot of custom code.

sonu4578 committed:
Improving validation of task retries to handle None values (#42532)

* Improving validation of task retries to handle None values

* Updated the validation check to use the "or" logical operator

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

* Added a comment for the new validation check

* Update baseoperator.py

* Update taskinstance.py

---------

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

kaxil committed:
Fix deprecated stage names for Pre-commit (#42872)

Before
```
❯ pre-commit install
[WARNING] top-level `default_stages` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
pre-commit installed at .git/hooks/pre-commit
```

pierrejeambrun committed:
Fix ui lint pre-commit interactive mode (#42854)

gopidesupavan committed:
update k8s tests urllib3 retry config status_forcelist and allowed_methods (#42871)

kaxil committed:
Bump `blacken-docs` pre-commit (#42880)

Bump `blacken-docs` pre-commit from 1.18.0 -> 1.19.0

kaxil committed:
Exclude "not-ready" providers when building docs (#42873)

This fixes the issue we saw in https://apache-airflow.slack.com/archives/C03G9H97MM2/p1728492002477169

This PR excludes the providers that aren't ready yet for building docs.

gopidesupavan committed:
Move Hooks to Standard provider (#42794)

pierrejeambrun committed:
Fix dag warning documentation (#42858)

eladkal committed:
Fix issue generation for provider after folder restructure (#42883)

omkar-foss committed:
AIP-84 Migrate the public endpoint Get DAG to FastAPI (#42848)

* Migrate the public endpoint Get DAG to FastAPI

* Use proper name for test function

rawwar committed:
AIP-84 Migrate GET Dag Run endpoint to FastAPI (#42725)

* get dag_run init

* add serializer

* Merge branch 'main' of https://github.com/apache/airflow into kalyan/AIP-84/get_dag_run

* add types

* add test

* working tests

* add note to DagRunResponse

* add note

* add test to test non Null note

* Update airflow/api_fastapi/views/public/dag_run.py

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>

* Update airflow/api_fastapi/views/public/dag_run.py

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>

* Merge branch 'main' of https://github.com/apache/airflow into kalyan/AIP-84/get_dag_run

* add 404 test

---------

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>

Commit:
add majorosdonat as bosch user (#42890)

Co-authored-by: Majoros Donat (XC-DX/EET2-Bp) <donat.majoros2@hu.bosch.com>

vincbeck committed:
Move the session auth backend to FAB auth manager (#42878)

vincbeck committed:
Move user and roles schemas to fab provider (#42869)

dstandish committed:
Add support for run conf to backfill (#42865)

This actually does a little bit more.  It changes the backfill create endpoint to take a json payload instead of just query params.  This is just easier because we can use the backfill schema as the schema of the request body.

One thing that is maybe weird is I add a decorator to translate the request body to the kwargs in the endpoint function.  The main motivator here was for compatibility with the requires_access_dag decorator, which doesn't check request body.

pierrejeambrun committed:
Fix main (#42903)

dstandish committed:
Fix main: js/types/api-generated.ts (#42906)

gopidesupavan committed:
mark TestGKEStartKueueInsideClusterOperator tests with flaky decorator (#42916)

dstandish committed:
Remove dag.run() method (#42761)

This method uses Backfill internally. Before we can remove BackfillJobRunner, we need to remove DAG.run. But before we can remove DAG.run, we need to update some old tests that use it. So this is the first step towards removing BackflilJobRunner.

There were some very old tests that came from airflow github issue 1225. These appeared to test the scheduler but really they tested the backfill job runner. Just to be cautious, I kept most of them rather than remove (which probably would have been fine since they essentially tested code that we'll be removing). As appropriate I either changed them to run on dag.test or scheduler. The ones dealing with ignore first depends on past will have to be added back when that functionality is implemented in new backfill.

dstandish committed:
Add test for behavior for paused backfill (#42837)


-----
Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

simi committed:
Use url_from_endpoint inside HttpHook. (#42785)

kaxil committed:
Fix typo in Breeze (#42919)

A couple of minor typos due to newlines

R7L208 committed:
Update airflow/providers/databricks/hooks/databricks_sql.py

Co-authored-by: Wei Lee <weilee.rx@gmail.com>

R7L208 committed:
Update airflow/providers/databricks/hooks/databricks_sql.py

Co-authored-by: Wei Lee <weilee.rx@gmail.com>

infused-kim committed:
Docs: Add templating info to TaskFlow tutorial (#42887)

Rasnar committed:
Add possibility to override the conn type for Druid (#42793)

* Add possibility to override the conn type for Druid

Minor fix, which allows to use the schema which are specified in
theschema rather than `http` as default. In the same time it doesn't
changethe logic as any conn_type can be selected. Intuitevely it's
expectedthat anything specified in `schema` field will actually take
precedencein the building the desired url.

* Add druid endpoint connection from another PR

* Fix missing scheme in test

* Set schema to None where it's unused

Even though we don't need it directly set, by default the mock will set
it to an internal object, thus we need to override it to None.

---------

Co-authored-by: Oleg Auckenthaler <github.sitcom838@passmail.net>

rawwar committed:
fix: HttpSensorTrigger to include `method` when serializing (#42925)

pierrejeambrun committed:
AIP-84 Fix dag display name search (#42863)

* Fix dag display name search

* Fix CI

gopidesupavan committed:
Check _is_canary_run/pr condition in is_legacy_ui_api_labeled method (#42921)

* check _is_canary_run condition in is_legacy_ui_api_labeled method

* include pr check in is_legacy_ui_api_labeled

ashb committed:
Allow python 3.12 for the breeze release commands (#42936)

I'm not 100% sure why we added this check in the first place, but it doesn't
seem to be needed anymore (I've tested things locally with this removed and it
all seems to behave itself)

bbovenzi committed:
Render errors when getting a list of dags (#42897)

* Render errors when getting a list of dags

* Restore axios, prettierignore pnpm-store

* Add pnpm-store to prettier ignore

kaxil committed:
Remove the referrer from Webserver to Scarf (#42901)

This will make sure we don't receive any information about the Webserver URL sending the info like the number of plugins and such.

vlieven committed:
Split providers out of the main "airflow/" tree into a UV workspace project (#42505) (#42624)

This is only a partial split so far. It moves all the code and tests, but
leaves the creation of `core/` to a separate PR as this is already large
enough.

In addition to the straight file rename the other changes I had to make here
are:

- Some mypy/typing fixes.

  Mypy can be fragile about what it picks up when, so maybe some of those
  changes were caused by that. But the typing changes aren't large.

- Improve typing in common.sql type stub

  Again, likely a mypy file oddity, but the types should be safe

- Removed the `check-providers-init-file-missing` check

  This isn't needed now that airflow/providers shouldn't exist at all in the
  main tree.

- Create a "dev.tests_common" package that contains helper files and common
  pytest fixtures

  Since the provider tests are no longer under tests/ they don't automatically
  share the fixtures from the parent `tests/conftest.py` so they needed
  extracted.

  Ditto for `tests.test_utils` -- they can't be easily imported in provider
  tests anymore, so they are moved to a more explicit shared location.

In future we should switch how the CI image is built to make better use of UV
caching than our own approach as that would remvoe a lot of custom code.

Co-authored-by: Ash Berlin-Taylor <ash@apache.org>
Co-authored-by: Ryan Hatter <25823361+RNHTTR@users.noreply.github.com>

gopidesupavan committed:
increase backoff_factor and add try/catch in k8s tests (#42940)

potiuk committed:
Consistent python version checks and troubleshooting (#42944)

Follow up after #42766 and #42936.

* We do not have to check for minimum Python version for Python 3.9
  any more (as we do not support 3.8 any more)
* While Python 3.12 is not yet fully supported by Apache Beam, we
  should still not allow it for releasing providers, but all other
  commands should support Python 3.12
* When you had pre-commit installed with Python 3.8 before, various
  errors might appear when running pre-commit. Troubleshooting
  was added quoting the errors and explaining what to do.

Commit:
Update README.rst (#42941)

grammar update for clarity 😄 👍

dstandish committed:
Add a "backfill create" command (#42922)

I create a new command group "backfill" for management of backfills. The first action is "create" which creates a backfill. Some others may follow such as pause / cancel.

sfc-gh-jdu committed:
Fix SNOWFLAKE_CONN_ID and DAG_ID in Snowpark system tests (#42952)

kaxil committed:
Fix broken links in Release Management docs (#42958)

These links were pointing to the wrong location. This PR fixes it

gopidesupavan committed:
Restrict looker-sdk version 24.18.0 and microsoft-kiota-http 1.3.4 (#42954)

* restrict looker version 24.18.0

* update microsoft provider deps with microsoft-kiota-http

* update gh issue ids to provider yaml files

* ignore type in applfowhook

* ignore type in applfowhook

* ignore type in applfowhook

gopidesupavan committed:
upgrade trove-classifiers (#42950)

Aakcht committed:
Chart: fix VCT for scheduler in local and persistent mode (#42946)

peloyeje committed:
fix: use instance base_container_name to fetch logs on trigger_reentry (#42960)

kgw7401 committed:
fix: 03_contributors_quick_start_docs (#42927)

* Change code-block position

kaxil committed:
Add skeleton project for task-sdk (#42904)

closes https://github.com/apache/airflow/issues/42856

This PR adds a skeleton project for [AIP-72 Task SDK](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-72+Task+Execution+Interface+aka+Task+SDK)

It creates this new project, integrates them to breeze & runs it on CI.

The code & tests are just placeholders at this point, we will start adding things needs for DAG Authors in separate PRs.

kaxil committed:
Upgrade Trove classifier to `2024.10.12` (#42961)

Test was failing: https://github.com/apache/airflow/actions/runs/11306055404/job/31446236535

```
pre-commit hook(s) made changes.
If you are seeing this message in CI, reproduce locally with: `pre-commit run --all-files`.
To run `pre-commit` as part of git workflow, use `pre-commit install`.
All changes made by hooks:
diff --git a/pyproject.toml b/pyproject.toml
index 844e1b6..5e9f587 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,7 +30,7 @@ requires = [
     "pluggy==1.5.0",
     "smmap==5.0.1",
     "tomli==2.0.2; python_version < '3.11'",
-    "trove-classifiers==2024.10.11",
+    "trove-classifiers==2024.10.12",
 ]
 build-backend = "hatchling.build"
```

kaxil committed:
Add missed brackets for our dev script for Spell checks (#42965)

A nit PR while I was looking at our doc builds

dabla committed:
FIX: Don't raise a warning in ExecutorSafeguard when execute is called from an extended operator (#42849)

* refactor: Don't raise a warning when execute is called from an extended operator, as this should always be allowed.

* refactored: Fixed import of test_utils in test_dag_run

---------

Co-authored-by: David Blain <david.blain@infrabel.be>

potiuk committed:
Upgrade Helm Chart dependencies to latest released (#42816)

jason810496 committed:
Fix PythonOperator DAG error when DAG has hyphen in name (#42902)

potiuk committed:
fIX handling removal of dependencies (#42967)

There was a problem with CI image builds when cache has been
disabled - they still used "pre-cached" main version (unlike
PROD builds). This PR synchronizes the behaviour between CI
and PROD builds.

For removing dependencies, you need to set both:

* `disable docker cache` label
* incerase DEPENDENCIES_EPOCH_NUMBER in `Dockerfile.ci`

Comments and documentation in both places has been updated to
reflect it.

Since documentation for labels has been updated,
Part of this PR is to improve the description of possible
labels that could be used during the build.
The description grew from a small number of labels to a
"wall of text" that was difficult to read. This PR reformats it in the
form of table that makes it far easier to see actions that the
maintainer can do and what labels should be set for each of them.

gopidesupavan committed:
mark test_task_workflow_trigger_success as flaky (#42972)

AryanK1511 committed:
Create a User Settings button with light/dark mode toggle as a menu item (#42964)

rawwar committed:
AIP-84 Migrate delete Dag Run endpoint to FastAPI (#42910)

* delete dag_run init

* delete dag_run init

* add delete dag run endpoint

* add tests for delete dag run

michaeljs-c committed:
Make datStats endpoint dag_ids parameter optional (#42955)

* Make datStats dag_id parameter optional with pagination

* move pagination out of sql query

* tidy

---------

Co-authored-by: Michael Smith-Chandler <mjsmithandler@gmail.com>

gopidesupavan committed:
trove classifier upgrade (#42979)

vincbeck committed:
Change directory used by simple auth manager to store generated passwords (#42860)

* Change directory used by simple auth manager to store generated passwords

* Update airflow/auth/managers/simple/simple_auth_manager.py

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

---------

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

gopidesupavan committed:
Disable flaky mssql based integration tests (#42811)

* quarantined flaky mssql integration tests

* disable mssql from ci integration checks

dstandish committed:
Remove BackfillJobRunner class (#42943)

olegkachur-e committed:
Add early job_id xcom_push for google provider Beam Pipeline operators (#42982)

- To let GCP Beam Sensor operators 'sense' the pipeline changes,
by having dataflow job_id been xcom_push as soon as it available.

Related issue: https://github.com/apache/airflow/issues/30007.

Co-authored-by: Oleg Kachur <kachur@google.com>

dstandish committed:
Add logic to mark backfills as complete (#42683)

Periodically check for backfills that should be marked as complete.

rawwar committed:
fix mypy check failure on main (#42976)

* fix mypy error

* Apply suggestions from code review

* Update conftest.py

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

---------

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>
Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

gopidesupavan committed:
mark test_setup_constraint_mapped_task_upstream_removed_and_success as flaky (#42997)

eladkal committed:
Update providers metadata 2024-10-14 (#42995)

ephraimbuddy committed:
Update json schema pre-commit to have draft7 schema in file (#43005)

We request this file each time the CI is running pre-commit and sometimes
it fails with 403 error. This PR adds the draft7 schema file locally for use
by pre-commit to avoid this error

pierrejeambrun committed:
AIP-84 Patch Variable (#42929)

ephraimbuddy committed:
Commit the session between writing and deletion of RTIF (#42928)

* Use different sessions in writing and deletion of RTIF

Previously, this was how it was done, but now,
a session was used for both the writing and deletion of RTIF,
which we suspect caused StaleDataError. The related PR: https://github.com/apache/airflow/pull/38565

This PR brings back the old behaviour of using different sessions for writing/deleting RTIFs

* fixup! Use different sessions in writing and deletion of RTIF

* add test and use flush

rawwar committed:
Refactor FastApi Dag and DagRun endpoints tests (#42949)

* wip - failing

* update tests

* update

* update tests

* update tests

* fix

* fix

* fix mypy error

* fix

kaxil committed:
Minor updates in UI contributing docs (#43013)

pierrejeambrun committed:
AIP-84 post variable (#42948)

dabla committed:
Feature: Added event_handler parameter in MSGraphAsyncOperator (#42539)

* refactor: Added parameter in MSGraphAsyncOperator to allow overriding default event_handler

* docs: Added docstring for event_handler parameter in MSGraphAsyncOperator

* refactor: Fixed TestMSGraphAsyncOperator

* refactor: Check if event is not None

* refactor: Register the TextParseNodeFactory and JsonParseNodeFactory so error messages get handled correctly in RequestAdapter

* refactor: Reorganized import for TestMSGraphAsyncOperator

* refactor: Added missing kiota-serialization packages in azure provider

* refactor: Updated provider dependencies

* refactor: Reorganized import of TestKiotaRequestAdapterHook

* refactor: Downgraded version of json kiota serialization

* refactor: Updated provider dependencies

* refactor: Put import of Context in TYPE_CHECKING block

* refactor: Fixed lookup of tenant-id

* refactor: Fixed kiota serialization dependencies to 1.0.0 to avoid pendulum dependency issues for backward compatibility

* refactor: Updated provider dependencies

* refactored: Fixed import of test_utils in test_dag_run

---------

Co-authored-by: David Blain <david.blain@infrabel.be>

dirrao committed:
uv version bump to 0.4.20 (#42905)

dirrao committed:
Removed deprecated Chainable type from BaseOperator (#42776)

* Removed deprecated Chainable type from BaseOperator

* Chainable word removed from spelling_wordlist.txt

* Chainable word removed from spelling_wordlist.txt

* Chainable word removed from spelling_wordlist.txt

* news fragment added

rawwar committed:
pin in  providers (#43001)

potiuk committed:
Move tests_common from "dev" to top-level. (#42985)

* Move tests common without changes

* Fix docstrings in tests_common

* Move tests_common from "dev" to top-level.

Follow-up after #42505 fixing teething problem with tests_common.

Originally in #42505 common test code was moved to "dev" folder, but
the "dev" folder is really dedicated to "build" scripts and the
problem with moving "tests_common" to the folder was that the
whole "dev" folder is replaced (for non-committer PRs) with
the content from the target branch.

This is done for security reasons, because we can accidentally
use any of the scripts from dev in the CI build scripts and we
might not notice, which will open us to a security issue where
a file in "dev" coming from PR could be accidentally executed
during the "pull_request_target" workflow - which would expose
our secrets and GitHub Package write permissions to a
contributor coming from a fork.

This change moves the files, fixes pre-commit specification and
docs, also fixes a number of "doc" issues detected by "ruff" in
the tests_common folder as they were detected after the move.
The tests_common folder is added to folders mounted when
breeze is executed with local folders mounted (in order to
avoid accidental mounting of randomly generated files to
inside the breeze container).

All imports for the common tests were updated to reflect this
move.

uranusjr committed:
Add AssetActive model (#42612)

omkar-foss committed:
Migrate the public endpoint Delete DAG to FastAPI (#42914)

* Migrate the public endpoint Delete DAG to FastAPI

* Refactor tests

amoghrajesh committed:
Clarifying PLUGINS_FOLDER permissions by DAG authors (#43022)

dirrao committed:
kubernetes package version bump to 31.0.0 (#42907)

saucoide committed:
Make google provider pyarrow dependency explicit (#42996)

* Make google provider pyarrow dependency explicit

The provider already depends on it directly but it was not listed in provider.yaml

* Add comment to the change

* rerun hooks

luyangliuable committed:
Add search by dag_display_name_pattern on dag list page with rebase (#42896)

* Add search by `dag_display_name_pattern` on dag list page.

* refactor search parameter update logic

* Refactor SearchBar component and simplify debounce logic in DagsList.

* Refactor search input change handling for SearchBar and DagsList components

* Update airflow/ui/src/pages/DagsList/DagsList.tsx

Co-authored-by: Brent Bovenzi <brent.bovenzi@gmail.com>

* Refactor search parameter handling and improve type consistency

* Remove typo

* Add `LAST_DAG_RUN_STATE` to `SearchParamsKeys` and update filters

* Add LAST_DAG_RUN_STATE to SearchParamsKeys and update filters

* Fix missing change for add SearchBar component by removing forwardRef and adding debounced search logic.

* minor change for SearchBar and DagsFilters components

* Optimize imports and improve formatting across components

* refactor: move query options from autogenerated useDagServiceGetDags to DagsList.

* Fix formatting.

---------

Co-authored-by: Brent Bovenzi <brent.bovenzi@gmail.com>

vincbeck committed:
Deprecate session auth backend (#42909)

kgw7401 committed:
fix: Change CustomSecurityManager method name (#43034)

* oauth_user_info -> get_oauth_user_info

potiuk committed:
Update to 0.4.21 of UV (#43032)

Commit:
Add retry on error 502 and 504 (#42994)

* Add retry on error 502 and 504

* fix mypy findings

* Add pytest

* Convert response code to HTTPStatus

* Add docs to retriable exception

* extend docs for AirflowHttpException

* Fix syntax and typos

* fix pytest

* fix static checks

* fix some static checks

* Fix ruff

* fix pre-commit

---------

Co-authored-by: Majoros Donat (XC-DX/EET2-Bp) <donat.majoros2@hu.bosch.com>

bugraoz93 committed:
AIP-84 Migrate get connections to FastAPI API #42571 (#42782)

* Make SortParam parent for Model Specific SortParams, Include get connections endpoint to fastapi

* Change depends() method regular method in SortParam due to parent class already have abstract

* Remove subclass, get default order_by from primary key, change alias strategy for backcompat

* pre-commit hooks

* Dynamic return value of SortParam generated within openapi specs and removed unnecessary attribute mapping keys

* Include connection_id to attr_mapping again

* Dynamic depends with correct documentation

* Add more tests

---------

Co-authored-by: pierrejeambrun <pierrejbrun@gmail.com>

jroachgolf84 committed:
Make `RedshiftDataOperator`  handle multiple queries (#42900)

* Adding updates to RedshiftDataOperator and unit tests

* Adding updates to RedshiftDataOperator and unit tests

* Updating how statement_id is used

* Ruff formatting

* Fixing nit

kaxil committed:
Bump `uv` to `0.4.21` in other places (#43033)

Follow-up of #43032: Bumps uv to 0.4.21 (https://pypi.org/project/uv/0.4.21/)

mobuchowski committed:
require 1.2.1 common.compat for openlineage provider (#43039)

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

bbovenzi committed:
Detect system color theme and add active state to nav button (#43041)

* Detect system color theme and add active state to nav button

* Clean up navbutton props

bbovenzi committed:
Migrate health info to fastapi (#42938)

* Add health info to public fastapi

* Add test and fix TaskInstanceState name

rawwar committed:
Add upperbound to microsoft-kiota-abstractions (#43021)

* add upperbound to microsoft-kiota-abstractions

* add

* add comment about breaking change

mobuchowski committed:
fix path to providers dir for mount sources breeze flag (#43042)

CYarros10 committed:
Create Operators for Google Cloud Vertex AI Context Caching (#43008)

* Fix merge conflicts

* Fix documentation.

* Update return variables.

rawwar committed:
add lowerbount to requests-toolbelt and replace requests_toolbelt with requests-toolbelt (#43020)

bdsoha committed:
✨ Allow node_selector templating in KPO (#43051)

kaxil committed:
Rename views to routes for FastAPI apps (#43057)

This is part of the change I want to do for https://github.com/apache/airflow/pull/43015 but going to break it down to keep it manageable.

Since these are actual routes for APIs, renaming them to routes instead of views.

ferruzzi committed:
Follow-up SLA purge (#42808)

Co-authored-by: dstandish <dstandish@users.noreply.github.com>

kaxil committed:
Bump `uv` to `0.4.22` (#43056)

https://pypi.org/project/uv/0.4.22

I added `--no-sources` to our `uv pip install` command, without it, it fails with following error

```
error: Failed to build: `apache-airflow @ file:///dist/apache_airflow-3.0.0.dev0.tar.gz`
  Caused by: Failed to parse entry for: `local-providers`
  Caused by: Package is not included as workspace package in `tool.uv.workspace`
```
Example: https://github.com/apache/airflow/actions/runs/11354720505/job/31582868760#step:11:2896

uranusjr committed:
Add 'name' and 'group' to public Asset class (#42812)

uranusjr committed:
Remove zombie ti from its executor (#42932)

uranusjr committed:
Drop unneeded unique() call on SQL (#43064)

kaxil committed:
Allow building docs for not-ready providers (#43071)

The `not-ready` providers were skipped in https://github.com/apache/airflow/pull/42873 as we were releasing a batch of providers.

I have manually gone ahead and uploaded inventories for edge & standard providers so this is no longer a problem.

biswa-b committed:
Typo correction in 07_local_virtualenv.rst - "way" (#43072)

ferruzzi committed:
Enable explicit namespaces (#42951)

* The recent PR to reorganize the package requires a change to explicit namespaces for at least some IDEs (PyChasrm for sure, possibly others).

* static fixes

* Update providers/src/airflow/providers/__init__.py

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

* Remove duplicate line

* Rephrase relevant comments

* Add providers tests init file to autoapi_ignore

* typo

Co-authored-by: Shahar Epstein <60007259+shahar1@users.noreply.github.com>

---------

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>
Co-authored-by: Shahar Epstein <60007259+shahar1@users.noreply.github.com>

potiuk committed:
Replace Mateusz with Kalyan in the triage team (#43074)

kaxil committed:
Reorganize ``api_fastapi`` folder into apps (#43062)

uranusjr committed:
Tweak AssetAlias to match Asset for AIP-74 additions (#42814)

kaxil committed:
Fix pytest from working outside breeze (#43082)

This was missed in https://github.com/apache/airflow/pull/42985 . Without this `airflow.providers.__path__` had 2 registered paths:

```
['/Users/kaxilnaik/Documents/GitHub/astronomer/airflow/providers/src/airflow/providers',
 '/Users/kaxilnaik/Documents/GitHub/astronomer/airflow/providers']
```

This prevents the tests from running outside of breeze and we get the following error:

```
ERROR tests/core/test_settings.py::test_usage_data_collection_disabled[true-True-True] - airflow.exceptions.AirflowConfigException: ("The provider apache-airflow-providers-src-airflow-providers-amazon is attempting to contribute configuration section aws that has already been added before. The source of it: apache-airflow-providers-amazon. This is forbidden. A provider can only add new sections. It cannot contribute options to existing sections or override other provider's configuration.", <class 'UserWarning'>)
```

We get this error because the *Providers Manager* uses `airflow.providers.__path__` to register providers. Because we have 2 paths, it registers the same provider twice leading two the above error.

https://github.com/apache/airflow/blob/75b22940ac4d36c31380669da2aa32fe46d70d32/airflow/providers_manager.py#L662

Example registration:
```
('apache-airflow-providers-src-airflow-providers-yandex',
  {'yandex': {'description': 'This section contains settings for Yandex Cloud ', ...

('apache-airflow-providers-yandex',
  {'yandex': {'description': 'This section contains settings for Yandex Cloud '
```

This wasn't a problem in breeze as it sets `AIRFLOW_SOURCES` env var in Dockerfile
https://github.com/apache/airflow/blob/75b22940ac4d36c31380669da2aa32fe46d70d32/scripts/docker/entrypoint_ci.sh#L24

shahar1 committed:
Fix typo in ``pyproject.toml`` (#43077)

dstandish committed:
Remove DAG.get_num_active_runs (#43067)

We don't need this function.  There's already an almost-identical function on DagRun that we can use, namely DagRun.active_runs_of_dags.

Also simplified that one.

pierrejeambrun committed:
Fix UI lint pre-commit hook (#43086)

kaxil committed:
Fixed failing static checks (#43087)

For some reasons https://github.com/apache/airflow/pull/43077 passed but main was failing. This PR fixes it.

AutomationDev85 committed:
Added task_instance_mutation_hook for mapped operator index 0 (#42661)

* Added task_instance_mutation_hook for mapped operator index 0

* Added unit test

---------

Co-authored-by: Marco Küttelwesch <marco.kuettelwesch@de.bosch.com>

potiuk committed:
Fix selective checks when only pyproject.toml changes (#43088)

In case only pyproject.toml changes we should automatically build
CI image as we need to run full set of static checks - including
mypy and other tools.

This change triggers ci-image for such case and it also fixes
a subtle bug where "pyproject_toml_changed" was always set
to true (but nothing depended on it, because we are checking if
build dependencies changed in pyproject.toml to trigger full build
as most of the build logic and dependencies is in hatch_build.py

The change first now attempts to see if the pyprject.toml
changed at all (i.e. if pyprooject.toml is in the list of files
that are changed in the commit) and only then reads the content
of old and new version of pyproject.toml to compare them. This
will make selective checks a bit faster as it won't run two git
commands when pyproject.toml has not changed.

Also tests_common are now added to "all sources" group which
should trigger tests when only tests_common change.

josix committed:
fix(providers/mongo): prevent applying lower method on boolean field (#43024)

gopidesupavan committed:
add kubernetes_conn_id to templated fields (#42786)

rawwar committed:
Add min version to cloudpickle (#43066)

* pin cloudpickle >= 3.0.0

* pin cloudpickle >= 2.1.1 as apache-beam requires it to be ~=2.2.1

bbovenzi committed:
Initialize dashboard page with health (#43090)

jscheffl committed:
Update constraints to broken canary (#43095)

kaxil committed:
Fix flaky `test_get_dags` in FastAPI routes (#43100)

gyandeeps committed:
SSHHook expose auth_timeout parameter (#43048)

* SSHHook expose auth_timeout parameter

* add tests

WSHoekstra committed:
vertex ai training operators: add display_name to rendered fields (#43028)

* vertex ai training operators: add display_name to rendered fields

* fix validate-operators-init static checks

---------

Co-authored-by: Walter Hoekstra <walterhoekstra@bol.com>
Co-authored-by: Shahar Epstein <60007259+shahar1@users.noreply.github.com>

jabbera committed:
Print the key name when max_length is exceeded (#43061)

* Print the key name when max_length is exceeded

* Fix tests

pierrejeambrun committed:
Revert "Fix flaky `test_get_dags` in FastAPI routes (#43100)" (#43108)

pierrejeambrun committed:
Ensure stable secondary ordering (#43085)

kaxil committed:
Add simple "Task Execution" API server (#43015)

closes https://github.com/apache/airflow/issues/43009

The Task Execution API is created as a separate FastAPI APP so that it allows us to hook CLI and selectively run the "Execution API" with the rest of Airflow API (UI/public). This is so that users can scale the APIs separately if needed as mentioned in #43009

rawwar committed:
Add ClientConnectorError to be a retryable error in databricks provider (#43091)

closes #43080

Bowrna committed:
passing the filetype for SlackAPIFileOperator (#43069)

sfirke committed:
chore(docs): add required import of BranchDayOfWeekOperator (#43053)



---------

Co-authored-by: Shahar Epstein <60007259+shahar1@users.noreply.github.com>

yangyulely committed:
Update unit_tests.rst links error (#43114)

eilon246810 committed:
added MultipleFilesWebHdfsSensor (#43045)

kaxil committed:
Bump to `0.4.23` to fix a uv bug (#43121)

Now that https://github.com/astral-sh/uv/issues/8236 is fixed (a bug in uv 0.4.22), we can remove `--no-sources` (used in `uv pip install`) introduced in https://github.com/apache/airflow/pull/43056

bbovenzi committed:
Clean up some eslint rules to better fit what we do (#43093)

kaxil committed:
Fixed failing static checks & provider tests (#43122)

rawwar committed:
Add min version to python-ldap (#43104)

* add min version to python-ldap

* add min version to python-ldap

Lee-W committed:
feat(providers/fab): Use asset in common provider (#43112)

kaxil committed:
Add `bugraoz93` in the triage team (#43124)

Commit:
make new exceptions databricks specific

Commit:
make new exceptions databricks specific

Commit:
accidentally removed necessary changes when resolving conflicts

Commit:
remove redefinition of mock_workflow_run_metadata

Commit:
merge main into this branch bc rebase failed

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

Lee-W commented on providers/src/airflow/providers/databricks/hooks/databricks_sql.py:
```diff
@@ -180,6 +196,7 @@ def run(
         self,
         sql: str | Iterable[str],
         autocommit: bool = ...,
+        execution_timeout: timedelta | None = ...,
```
It would be better for us to add it as the last argument or keyword only argument to avoid breaking change

Lee-W commented on providers/tests/databricks/hooks/test_databricks_sql.py:
```diff
@@ -236,67 +298,51 @@ def test_query(
-                match="""Returning a raw `databricks.sql.Row` object is deprecated. A namedtuple will be
+    connections = []
+    cursors = []
+    for index in range(len(cursor_descriptions)):
```
```suggestion
    for index, cursor_description in enumerate(cursor_descriptions):
```

Lee-W commented on providers/tests/databricks/hooks/test_databricks_sql.py:
```diff
@@ -236,67 +298,51 @@ def test_query(
+        conn = mock.MagicMock()
+        cur = mock.MagicMock(
+            rowcount=len(cursor_results[index]),
+            description=get_cursor_descriptions(cursor_descriptions[index]),
```
```suggestion
            description=get_cursor_descriptions(cursor_description),
```

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

R7L208 committed:
Update providers/tests/databricks/hooks/test_databricks_sql.py

Co-authored-by: Wei Lee <weilee.rx@gmail.com>

R7L208 committed:
Update providers/tests/databricks/hooks/test_databricks_sql.py

Co-authored-by: Wei Lee <weilee.rx@gmail.com>

Commit:
make execution_timeout last arg and keyword arg

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent

R7L208 committed:
Merge branch 'main' into lorin/databricks-sql-operator-on_kill-equivalent
